app/models/agents/liquid_output_agent.rb

```diff
@@ -0,0 +1,175 @@
+        "expected_receive_period_in_days" => 2,
+        "content" => 'This is a Liquid template. Include variables from your last event, like {{this}} and {{that}}.',
+        "mime_type" => 'text/html',
+        "mode" => 'Last event in',
+        "event_limit" => '',
+      }
+    end
+
+    form_configurable :secret
+    form_configurable :expected_receive_period_in_days
+    form_configurable :content, type: :text
+    form_configurable :mime_type
+    form_configurable :mode, type: :array, values: [ 'Last event in', 'Merge events', 'Last X events']
+    form_configurable :event_limit
+
+    def working?
+      last_receive_at && last_receive_at > options['expected_receive_period_in_days'].to_i.days.ago && !recent_error_logs?
+    end
+
+    def validate_options
+      if options['secret'].present?
+        case options['secret']
+        when %r{[/.]}
+          errors.add(:base, "secret may not contain a slash or dot")
+        when String
+        else
+          errors.add(:base, "secret must be a string")
+        end
+      else
+        errors.add(:base, "Please specify one secret for 'authenticating' incoming feed requests")
+      end
+
+      unless options['expected_receive_period_in_days'].present? && options['expected_receive_period_in_days'].to_i > 0
+        errors.add(:base, "Please provide 'expected_receive_period_in_days' to indicate how many days can pass before this Agent is considered to be not working")
+      end
+    end
+
+    def receive(incoming_events)
+      return unless ['Merge events', 'Last event in'].include?(options['mode'])
+      memory['last_event'] ||= {}
+      incoming_events.each do |event|
+        case options['mode']
+        when 'Merge events'
+          memory['last_event'] = memory['last_event'].merge(event.payload)
+        else
+          memory['last_event'] = event.payload
+        end
+      end
+    end
+
+    def receive_web_request(params, method, format)
+      valid_authentication?(params) ? [liquified_content, 200, mime_type]
+                                    : [unauthorized_content(format), 401]
+    end
+
+    private
+
+    def unauthorized_content(format)
+      format =~ /json/ ? { error: "Not Authorized" }
+                       : "Not Authorized"
+    end
+
+    def valid_authentication?(params)
+      interpolated['secret'] == params['secret']
+    end
+
+    def mime_type
+      options['mime_type'].present? ? options['mime_type'] : 'text/html'
+    end
+
+    def liquified_content
+      template = Liquid::Template.parse(options['content'] || "")
+      template.render(data_for_liquid_template)
+    end
+
+    def data_for_liquid_template
+      case options['mode']
+      when 'Last X events'
+        events = received_events
+        events = events.where('events.created_at > ?', date_limit) if date_limit
+        events = events.limit(count_limit) if count_limit
+        events = events.to_a.map { |x| x.payload }
+        { 'events' => events }
```

Comment:
Nevermind on those ideas.... we do want a last event mode because we don't want a liquid template written against the last even to have to use ```{{events[0].name}}```.  Much better is to use ```{{name}}```.